### PR TITLE
Dev/transfer bbox with depth map

### PR DESCRIPTION
**Pull request by dmjoy**
This PR adds an algorithm implementation to the Core arrow for transferring detection object bounding boxes between two cameras with respect to a depth map.

### CODE REVIEW

File: arrows/ocv/transfer_with_depth_map.cxx [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+namespace kwiver {
+namespace arrows {
+namespace ocv {
+
+// ---------------------------------------------------------------------------
+transfer_with_depth_map::
+transfer_with_depth_map()
+  : src_camera_krtd_file_name( "" )
+  , dest_camera_krtd_file_name( "" )
+  , src_camera_depth_map_file_name( "" ) {
```
[the remark this answers is not in the document]
Comment (dmjoy) [reply arrows/ocv/transfer_with_depth_map.cxx]: Gotcha, should be fixed

File: arrows/ocv/tests/test_transfer_with_depth_map.cxx [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+using namespace kwiver::vital;
+
+path_t g_data_dir;
+
+static std::string src_cam_file_name = "src_camera.krtd";
+static std::string dest_cam_file_name = "dest_camera.krtd";
+
+// ----------------------------------------------------------------------------
+int main( int argc, char* argv[] ) {
```
[the remark this answers is not in the document]
Comment (dmjoy) [reply arrows/ocv/tests/test_transfer_with_depth_map.cxx]: Fixed the brace placement to be consistent with the style guide, and pared down lines to < 80 chars as well

**Comment by mattbrown11** [comment]
I'd like @mleotta to weigh in as well. I want to make sure we aren't adding something into the main KWIVER pool that is too narrowly defined, without the appropriate indication in the naming, for our DIVA problem. Longer term, having a general set of capabilities for moving detections from imagery into the world, and the likewise world to image transformations, will become relevant across all projects.

I'll list some variations that I can see becoming relevant in the near future for us to keep in mind as we structure this instance:
1) Don't have a depth map but assume people are X meters tall, so we can estimate the depth (with a degree of uncertainty).
2) Don't have a depth map but have a planar surface model of the world (more-relevant for downward looking imagery).
3) Don't have a fixed depth map, but can render one live from a provided meshed model.
4) Don't have a depth map, but the src/dst cameras are co-located, so depth isn't required to transform between them
5) For now, we are considering bounding boxes from mostly side-view imagery, therefore we assume the bottom center of the bounding box is what attaches to the world surface. However, from top-down views, you would want to unproject the center of the bounding box.
6) When projecting into the dst camera, you may also want to know if the detected object is occluded and do something with that information. This would typically require that you have the depth map for the dst camera and check to see if the object is located beyond what the depth map indicates.

Given the possible permutations of the above, I think it makes sense to separate the unprojection-from-src from the reprojection-into-dst. Maybe we define an unprojection vital algorithm for projecting detections from a camera into the world, populating the geo_point field, ideally with a covariance of certainty populated. Then, what we will be making here would be an OCV arrow for unprojecting using a depth map. Then, we would also be making a sprokit process that warps detections from one camera to another, which itself accepts the vital unprojection algorithm, and we would make an arrow for that process specifically for side-view imagery.

**Comment by dmjoy** [comment]
I've removed the explicit ocv dependency and moved the algorithm implementation to the core arrow instead of ocv as suggested by Matt L.

**Comment by linus-sherrill** [comment]
Is this ready to review or is there still work to be done?

**Comment by dmjoy** [comment]
Still a few minor things to tweak (renaming some functions), though I think the content could still be reviewed

**Comment by dmjoy** [comment]
Removing the draft flag, as I believe this is ready for review.

@mattbrown11 I've renamed the `transfer_bbox_with_depth_map` function to `transfer_bbox_with_depth_map_stationary_camera`.  In the future, if there are other transfer algorithms that still rely on a depth map (e.g. from an aerial camera) we could add a configuration option to specify which to use.

**Comment by linus-sherrill** [comment]
@mattbrown11 Does this look good to you?

**Comment by mattbrown11** [comment]
Looks good to me. I think this is ready to merge.

**Comment by linus-sherrill** [comment]
Jenkins test this please

**Comment by mwoehlke-kitware** [comment]
This broke the build. `test_transfer_bbox_with_depth_map.cxx` requires an OpenCV header, but is not part of the OpenCV arrow. This needs to build with OpenCV disabled...

**Comment by dmjoy** [comment]
Ah, my mistake, I forgot to remove the OpenCV imports when I relocated this functionality from the OpenCV arrow to Core (they shouldn't be necessary anymore).  I'll can put together a new PR with the fix.

**Comment by mwoehlke-kitware** [comment]
>  I'll can put together a new PR with the fix.

Appreciated, thanks. (Yeah, I was just noticing those don't seem to actually be used. For now I've worked around it by just commenting them out, but obviously we should fix the repo :slightly_smiling_face:.)

**Pull request merged**